### PR TITLE
ci: Attempt to fix some flaky Elixir unit tests

### PR DESCRIPTION
**Pull request by jamilbk**
Possibly fixes some of the flaky unit tests that crop up from time to time.

Refs #4635

### CODE REVIEW

File: elixir/apps/domain/test/support/mocks/openid_connect.ex [hunk 1]
```diff
@@ -91,7 +90,13 @@ defmodule Domain.Mocks.OpenIDConnect do
         "request_parameter_supported" => false
       }
 
-      Plug.Conn.resp(conn, 200, Jason.encode!(attrs))
+      # Process may not be alive in slow CI environments
```
Comment (jamilbk) [inline elixir/apps/domain/test/support/mocks/openid_connect.ex:93]: https://github.com/firezone/firezone/actions/runs/8907610107/job/24461787301?pr=4843

File: elixir/apps/web/test/support/acceptance_case.ex [hunk 2]
```diff
@@ -134,7 +134,8 @@ defmodule Web.AcceptanceCase do
     if env = System.get_env("E2E_DEFAULT_WAIT_SECONDS") do
       String.to_integer(env)
     else
-      2
+      # GitHub shared runners can be very slow during peak hours
```
Comment (jamilbk) [inline elixir/apps/web/test/support/acceptance_case.ex:137]: https://github.com/firezone/firezone/actions/runs/8725396789/job/23938263443

File: elixir/apps/domain/test/support/mocks/openid_connect.ex [hunk 3]
```diff
@@ -142,8 +159,14 @@ defmodule Domain.Mocks.OpenIDConnect do
         )
 
       conn = fetch_conn_params(conn)
-      send(test_pid, {:request, conn})
-      Plug.Conn.resp(conn, 200, Jason.encode!(attrs))
+
+      # Process may not be alive in slow CI environments
+      if Process.alive?(test_pid) do
```
Comment (AndrewDryga) [inline elixir/apps/domain/test/support/mocks/openid_connect.ex:164]: This doesn't fix anything, the issue happens when Plug process is trying to read from Bypass process. So when the Plug process is terminated with the test process, the bypass can't finish reading the full response. And it doesn't matter if the response code is 503 or 401, has body or not, etc.

File: elixir/apps/web/test/support/acceptance_case.ex (hunk 2 above)
Comment (AndrewDryga) [reply elixir/apps/web/test/support/acceptance_case.ex:137]: @jamilbk it's already 5 seconds on CI: [ E2E_DEFAULT_WAIT_SECONDS in ](https://github.com/firezone/firezone/blob/0c7c96dd074d1bde1b1c8f3855503365a154768a/.github/workflows/_elixir.yml#L279)

Comment (jamilbk) [reply elixir/apps/web/test/support/acceptance_case.ex:137]: 😢

**Pull request closed without merging**